app/src/main/java/org/eu/exodus_privacy/exodusprivacy/fragments/dialog/ExodusDialogFragment.kt

```diff
@@ -1,46 +1,94 @@
 import androidx.core.text.HtmlCompat
 import androidx.fragment.app.DialogFragment
 import androidx.fragment.app.viewModels
 import com.google.android.material.dialog.MaterialAlertDialogBuilder
 import dagger.hilt.android.AndroidEntryPoint
+import org.eu.exodus_privacy.exodusprivacy.ExodusUpdateService
+import org.eu.exodus_privacy.exodusprivacy.MainActivityViewModel
 import org.eu.exodus_privacy.exodusprivacy.R
 
 @AndroidEntryPoint
 class ExodusDialogFragment : DialogFragment() {
 
     private val TAG = ExodusDialogFragment::class.java.simpleName
-    private val exodusDialogViewModel: ExodusDialogViewModel by viewModels()
+    private val exodusDialogViewModel: MainActivityViewModel by viewModels()
+    private val permission = "android.permission.POST_NOTIFICATIONS"
+    val version = Build.VERSION.SDK_INT
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
 
-        exodusDialogViewModel.policyAgreement.observe(viewLifecycleOwner) {
-            this.dismiss()
+        exodusDialogViewModel.config.observe(viewLifecycleOwner) { config ->
+            if (config["privacy_policy"]?.enable!!) this.dismiss()
         }
     }
 
     override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
-        return MaterialAlertDialogBuilder(requireContext())
-            .setTitle(getString(R.string.warning_title))
+        // Initialize the config
+        exodusDialogViewModel.config.observe(this) {}
+        return MaterialAlertDialogBuilder(requireContext()).setTitle(getString(R.string.warning_title))
             .setMessage(
                 HtmlCompat.fromHtml(
-                    getString(R.string.warning_desc),
-                    HtmlCompat.FROM_HTML_MODE_COMPACT
+                    getString(R.string.warning_desc), HtmlCompat.FROM_HTML_MODE_COMPACT
                 )
-            )
-            .setPositiveButton(getString(R.string.accept)) { _, _ ->
+            ).setPositiveButton(getString(R.string.accept)) { _, _ ->
                 Log.d(TAG, "Permission to transmit data granted!")
                 exodusDialogViewModel.savePolicyAgreement(true)
-            }
-            .setNegativeButton(getString(R.string.reject)) { _, _ ->
+                Log.d(TAG, "Version is: $version")
+                if (version < 33) {
+                    startInitial()
+                } else {
+                    if (!isNotificationPermissionGranted()) {
+                        requestPermission()
+                    }
+                }
+            }.setNegativeButton(getString(R.string.reject)) { _, _ ->
                 activity?.finish()
+            }.create()
+    }
+
+    private val requestPermissionLauncher = registerForActivityResult(
+        ActivityResultContracts.RequestPermission()
+    ) { isGranted: Boolean ->
+        if (isGranted) {
+            exodusDialogViewModel.saveNotificationPermissionRequested(true)

```

Comment:
Why use condition to execute same method ?

Reply:
Good point. I changed that.